s3/filestorage.go

```diff
@@ -423,9 +423,14 @@ func (s *SimpleStorageService) UploadArtifact(
 	artifact io.Reader,
 	contentType string,
 ) error {
+	maxImageSize := config.Config.GetInt64(dconfig.SettingAwsS3MaxImageSize)
+	minMultipartSize := int64(10 * 1024 * 1024) // 10MiB (must be at least 5MiB)
 	// NOTE: This size along with the 10000 part limit sets the ultimate
-	//       limit on the upload size. (currently at ~97.5GiB)
-	const multipartSize = 10 * 1024 * 1024 // 10MiB (must be at least 5MiB)
+	//       limit on the upload size.
+	multipartSize := maxImageSize / 1024
+	if multipartSize < minMultipartSize {
+		multipartSize = minMultipartSize
+	}
```

Comment:
Can we calculate the upload partition size once and store it in the `SimpleStorageService`? Can we also round the page size to nearest MiB that will fit maxImageSize in 10,000 pages?
```go
mib := 1024 * 1024
// partSize must be at least 5MiB
partSize := max(5 * mib, ((maxImageSize-1) / (10000 * mib) + 1) * mib)
```